### PR TITLE
statistics: fix empty region count when resuming

### LINKED ISSUE #7008: Statistics: EmptyRegion count is incorrect after the cluster is paused and resumed
## Bug Report

img_v2_7d85c37a-cdb7-4f53-84d9-08b47492324g

### What did you do?
as title

### What did you expect to see?
EmptyRegion is very few
### What did you see instead?
EmptyRegion equals region number

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: Close #7008 #5273

### What is changed and how does it work?
We only save `meta` when save regions, so the approximate size of the original region is zero when restarting and `Inherit` is useless.

```commit-message
```

### Check List

Tests

- Manual test (add detailed scripts or steps below)
image

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/statistics/region_collection.go [hunk 1]
```diff
@@ -198,7 +198,9 @@ func (r *RegionStatistics) Observe(region *core.RegionInfo, stores []*core.Store
 			return false
 		}(),
 		LearnerPeer: len(region.GetLearners()) > 0,
-		EmptyRegion: region.GetApproximateSize() <= core.EmptyRegionApproximateSize,
+		// When cluster resumes, the region size may be not initialized, but region heartbeat is send.
```
Comment (rleungx) [inline pkg/statistics/region_collection.go:201]: If the region size is not initialized, will it inherit from the original one?

**Review by rleungx — APPROVED** [review]

File: pkg/statistics/region.go [hunk 2]
```diff
@@ -59,7 +59,7 @@ func (s *RegionStats) Observe(r *core.RegionInfo) {
 	approximateKeys := r.GetApproximateKeys()
 	approximateSize := r.GetApproximateSize()
 	approximateKvSize := r.GetApproximateKvSize()
-	if approximateSize <= core.EmptyRegionApproximateSize {
+	if approximateSize == core.EmptyRegionApproximateSize {
 		s.EmptyCount++
 	}
 	s.StorageSize += approximateSize
```
Comment (Connor1996) [inline pkg/statistics/region.go:65]: approximateSize == 1 means empty, should we add it to storage size?

Comment (CabinfeverB) [reply pkg/statistics/region.go:65]: Do u mean current storage size? If so, it is calculated from the used size of store

Comment (Connor1996) [reply pkg/statistics/region.go:65]: I mean 
```go
if approximateSize > core.EmptyRegionApproximateSize {
   s.StorageSize += approximateSize
}
```

File: pkg/statistics/region_collection.go [hunk 3]
```diff
@@ -198,15 +198,17 @@ func (r *RegionStatistics) Observe(region *core.RegionInfo, stores []*core.Store
...
+		// So use `==` here.
+		EmptyRegion: region.GetApproximateSize() == core.EmptyRegionApproximateSize,
```
Comment (Connor1996) [inline pkg/statistics/region_collection.go:203]: can we extract a function `IsEmptyRegion()` for `regionInfo`

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**